app/controllers/posts_controller.rb

```diff
@@ -4,10 +4,21 @@ class PostsController < ApplicationController
   def index
```

Comment:
:red_circle: [REQUIRED] I'm having an issue with the API response from your code. When I hit the API endpoint, it's not returning a list of posts in JSON format. Instead, I'm getting the response "You need to sign in or sign up before continuing."

I've attached an image that illustrates the issue I'm facing.

![update](https://github.com/fredo509/Trading-forum/assets/26736582/2588363b-c743-40ca-9bb9-1cf2384327f0)


Can you please take a look at the code and see if you can figure out what's going wrong?

HINT :bulb: :  To fix this, you need to add authentication middleware to your API. This middleware will check to see if the user is authenticated before they are allowed to access the API( **strongly recommended approach** ).

Alternatively, you can skip the token verification for all requests of JSON type. However, this will require you to disable authorization for this to work. I believe that the first option is the best approach.